openwisp_notifications/tests/test_widget.py

```diff
@@ -54,3 +53,27 @@ def test_notification_relative_link(self):
         self.assertEqual(
             data_location_value, _get_object_link(notification, 'target', False)
         )
+
+    def test_notification_dialog(self):
+        self.login()
+        self.notification_options.update(
+            {'message': 'Test Message', 'description': 'Test Description'}
+        )
+        notification = self._create_notification().pop()[1][0]
+        self.web_driver.find_element(By.ID, 'openwisp_notifications').click()
+        time.sleep(4)
+        WebDriverWait(self.web_driver, 10).until(
+            EC.visibility_of_element_located((By.ID, f'ow-{notification.id}'))
+        )
+        self.web_driver.find_element(By.ID, f'ow-{notification.id}').click()
```

Comment:
The click on the notification element doesn't seem to be clicking properly. I'm not sure why this occurs, but when manually clicked, it has no issues.

Reply:
I think we can use that to add an additional CSS class to the notification element. If that class is present, the the logic for marking notification as read is skipped and it opens the notification dialog directly. We can also use a data attribute for this.

Reply:
I have tried a different approach to solving it, but is it okay to let it be `unread` even if the user has read that message?

Reply:
> I have tried a different approach to solving it, but is it okay to let it be `unread` even if the user has read that message?

No, it is not.